File: rust/src/internal/api/endpoint_auto_config.rs
// this file is @generated
use crate::{error::Result, models::*, Configuration};

pub struct EndpointAutoConfig<'a> {
    cfg: &'a Configuration,
}

impl<'a> EndpointAutoConfig<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Update an auto-config endpoint by providing endpoint details.
    pub async fn update(
        &self,
        app_id: String,
        endpoint_id: String,
        subscribe_in: SubscribeIn,
    ) -> Result<EndpointOut> {
        crate::request::Request::new(
            http1::Method::PUT,
            "/api/v1/app/{app_id}/endpoint/{endpoint_id}/auto-config",
        )
        .with_path_param("app_id", app_id)
        .with_path_param("endpoint_id", endpoint_id)
        .with_body_param(subscribe_in)
        .execute(self.cfg)
        .await
    }
}

File: csharp/Svix/ApiInternal/EndpointAutoConfig.cs
// this file is @generated
#nullable enable
using Microsoft.Extensions.Logging;
using Svix.Models;

namespace Svix.ApiInternal
{
    public class EndpointAutoConfig(SvixClient client)
    {
        readonly SvixClient _client = client;

        /// <summary>
        /// Update an auto-config endpoint by providing endpoint details.
        /// </summary>
        public async Task<EndpointOut> UpdateAsync(
            string appId,
            string endpointId,
            SubscribeIn subscribeIn,
            CancellationToken cancellationToken = default
        )
        {
            subscribeIn = subscribeIn ?? throw new ArgumentNullException(nameof(subscribeIn));
            try
            {
                var response = await _client.SvixHttpClient.SendRequestAsync<EndpointOut>(
                    method: HttpMethod.Put,
                    path: "/api/v1/app/{app_id}/endpoint/{endpoint_id}/auto-config",
                    pathParams: new Dictionary<string, string>
                    {
                        { "app_id", appId },
                        { "endpoint_id", endpointId },
                    },
                    content: subscribeIn,
                    cancellationToken: cancellationToken
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(UpdateAsync)} failed");

                throw;
            }
        }

        /// <summary>
        /// Update an auto-config endpoint by providing endpoint details.
        /// </summary>
        public EndpointOut Update(string appId, string endpointId, SubscribeIn subscribeIn)
        {
            subscribeIn = subscribeIn ?? throw new ArgumentNullException(nameof(subscribeIn));
            try
            {
                var response = _client.SvixHttpClient.SendRequest<EndpointOut>(
                    method: HttpMethod.Put,
                    path: "/api/v1/app/{app_id}/endpoint/{endpoint_id}/auto-config",
                    pathParams: new Dictionary<string, string>
                    {
                        { "app_id", appId },
                        { "endpoint_id", endpointId },
                    },
                    content: subscribeIn
                );
                return response.Data;
            }
            catch (ApiException e)
            {
                _client.Logger?.LogError(e, $"{nameof(Update)} failed");

                throw;
            }
        }
    }
}
